src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,32 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * The command to be executed. All commands are subclasses of this class.
+ */
+public abstract class Command {
+    protected TaskList taskList;
```

Comment:
Perhaps these class variables could be declared `private final`, to better abide by the guidelines of Java information hiding and encapsulation.

Reply:
I made this protected so that it's available within the package, but I agree with your comment. I'll make it private.